src/response.rs

```diff
@@ -85,15 +76,13 @@ impl Response {
     /// Example:
     ///
     /// ```
-    /// let resp = ureq::Response::new(401, "Authorization Required", "Please log in");
+    /// let resp = ureq::Response::new(401, "Authorization Required", "Please log in").unwrap();
     ///
     /// assert_eq!(resp.status(), 401);
     /// ```
-    pub fn new(status: u16, status_text: &str, body: &str) -> Self {
+    pub fn new(status: u16, status_text: &str, body: &str) -> Result<Response, Error> {
         let r = format!("HTTP/1.1 {} {}\r\n\r\n{}\n", status, status_text, body);
-        (r.as_ref() as &str)
-            .parse::<Response>()
-            .unwrap_or_else(|e| e.into())
+        (r.as_ref() as &str).parse::<Response>()
```

Comment:
You don't need the type annotations here I think.
```suggestion
        (r.as_ref() as &str).parse()
```

Reply:
Ah yes, let's try without!